roxie/ccd/ccdserver.cpp

```diff
@@ -2297,12 +2298,13 @@ class RecordPullerThread : public RestartableThread
     {
         if (eof)
             return false;
-        while (preload)
+        while (preload && !eof)
         {
-            const void * row;
+            const void * row = nullptr;
             {
                 CriticalBlock c(crit); // See comments in stop for why this is needed
-                row = inputStream->nextRow();
+                if (!eof)
```

Comment:
It probably isn't very speed critical, but 
```
    if (eof)
       break

``` might generate better code (and would avoid the need for the check on the while).  With the check on the while eof should probably be an atomic.

Reply:
Actually return false; rather than break;

Reply:
I changed it as suggested, then changed my mind and changed it back. I don't think you want to skip the eog processing?

You might argue that we should use a separate variable rather than eof to signify "stopped asynchronously"

Reply:
Fair enough.  eof, should probably be atomic then, otherwise compilers could avoid the retest within the critical section.

Reply:
@richardkchapman